debugfs/dump.c

```diff
@@ -97,61 +97,71 @@ static void fix_perms(const char *cmd, const struct ext2_inode *inode,
 		com_err(cmd, errno, "while setting times of %s", name);
 }
 
-static void dump_file(const char *cmdname, ext2_ino_t ino, int fd,
+static int dump_file(const char *cmdname, ext2_ino_t ino, int fd,
 		      int preserve, char *outname)
 {
 	errcode_t retval;
+  errcode_t retcode;
 	struct ext2_inode	inode;
 	char		*buf = 0;
 	ext2_file_t	e2_file;
 	int		nbytes;
 	unsigned int	got, blocksize = current_fs->blocksize;
 
 	if (debugfs_read_inode(ino, &inode, cmdname))
-		return;
+		return 1;
 
 	retval = ext2fs_file_open(current_fs, ino, 0, &e2_file);
 	if (retval) {
 		com_err(cmdname, retval, "while opening ext2 file");
-		return;
+		return retval;
 	}
 	retval = ext2fs_get_mem(blocksize, &buf);
 	if (retval) {
 		com_err(cmdname, retval, "while allocating memory");
-		return;
+		return retval;
 	}
 	while (1) {
 		retval = ext2fs_file_read(e2_file, buf, blocksize, &got);
-		if (retval)
+		if (retval) {
 			com_err(cmdname, retval, "while reading ext2 file");
+			break;
+		}
 		if (got == 0)
 			break;
 		nbytes = write(fd, buf, got);
-		if ((unsigned) nbytes != got)
-			com_err(cmdname, errno, "while writing file");
+		if ((unsigned) nbytes != got) {
+      com_err(cmdname, errno, "while writing file");
+      break;
```

Comment:
we should set retval to 1 or something positive, otherwise retval is still 0 and retcode would be 0 as well.

Reply:
done